src/openrct2/world/Sprite.cpp

```diff
@@ -607,51 +607,51 @@ void sprite_misc_explosion_flare_create(const CoordsXYZ& flarePos)
  *
  *  rct2: 0x006733B4
  */
-static void sprite_misc_explosion_flare_update(rct_sprite* sprite)
+static void sprite_misc_explosion_flare_update(ExplosionFlare* sprite)
 {
-    sprite->generic.Invalidate2();
-    sprite->generic.frame += 64;
-    if (sprite->generic.frame >= (124 * 64))
+    sprite->Invalidate2();
+    sprite->frame += 64;
+    if (sprite->frame >= (124 * 64))
     {
-        sprite_remove(&sprite->generic);
+        sprite_remove(sprite);
     }
 }
 
 /**
  *
  *  rct2: 0x006731CD
  */
-static void sprite_misc_update(rct_sprite* sprite)
+static void sprite_misc_update(SpriteBase* sprite)
 {
```

Comment:
Should this function be checking for pointer validity?

Reply:
I'm hoping that that is not required as the As template should mean that part of it is taken care of by the switch. None of the compilers are complaining about that so i think it is happening like that.

Reply:
Well i checked the assembly of the version generated on my PC. It did not work out that its already validated the type so not quite as good as I hoped.